src/Client/Components/Header/AppMainHeader.razor.cs

```diff
@@ -0,0 +1,62 @@
+﻿
+using InterviewApp.Shared.Models;
+using Microsoft.AspNetCore.Components;
+
+namespace InterviewApp.Client.Components.Header;
+
+public partial class AppMainHeader
+{
+    [Parameter]
+    public string? Title { get; set; }
+
+    [Parameter]
+    public Func<string, IEnumerable<InterviewQuestion>>? OnSearch { get; set; }
+
+    [Parameter]
+    public Func<string, object>? OnSearchText { get; set; }
+
+    [Parameter]
+    public Func<InterviewQuestion, object>? OnSearchValue { get; set; }
+
+    [Parameter]
+    public EventCallback OnSwitchTheme { get; set; }
+
+    [Parameter]
+    public EventCallback OnOpenSettings { get; set; }
+
+    [Parameter]
+    public EventCallback OnResetQuestions { get; set; }
+
+    [Parameter]
+    public EventCallback OnOpenReport { get; set; }
+
+    public void SearchText(string text)
+        => OnSearchText!.Invoke(text);
+
+    public void SearchValue(InterviewQuestion value)
+        => OnSearchValue!.Invoke(value);
```

Comment:
Rather then asserting not null, you should instead not call it if it's null.

```suggestion
    public void SearchText(string text)
        => OnSearchText?.Invoke(text);

    public void SearchValue(InterviewQuestion value)
        => OnSearchValue?.Invoke(value);
```